process/smartContract/process.go

```diff
@@ -464,14 +464,27 @@ func (sc *scProcessor) isSelfShard(address []byte) bool {
 	return sc.shardCoordinator.ComputeId(address) == sc.shardCoordinator.SelfId()
 }
 
+func safeSubUint64(a, b uint64) uint64 {
+	if a < b {
+		return 0
```

Comment:
Now, due to the refactoring below, is this still possible or is it an error / critical condition?

Reply:
this should be not more possible - however it is better to be safe than sorry, thus have this kind of protection.